### ISSUE TITLE
Link on the flakiness dashboard points to the wrong log

### ISSUE THREAD

**Issue by alexmarkov** [dart-lang/sdk#36070]
On the Dart VM flakiness dashboard, the link "7" (in CTE column) on the line

2019-02-28 07:47:09 | language_2/void/void_type_usage_test/final_local_do_while | dartk-optcounter-linux-release-x64 | CompileTimeError,Crash | CompileTimeError | 7 |   | 7 |   |   | 1 |  

goes to https://logs.chromium.org/logs/dart/buildbucket/cr-buildbucket.appspot.com/8920261011728247104/+/steps/test_results/0/logs/ignored_flaky_test_failure_logs/0 which has information about failed service/isolate_lifecycle_test test and doesn't mention language_2/void/void_type_usage_test.

/cc @whesse @athomas

**Comment by sortie** [comment]
Hi Alex,

The testing infrastructure only produces logs for tests whose outcome is different from the test's expectation. The test `language_2/void/void_type_usage_test/final_local_do_while` is supposed to produce a `CompileTimeError` which is why there's no log for that flaky outcome, since it's the case that's normally not interesting. The flakiness dashboard does not show what the expected behavior is, which leads to this confusion.

Bill, do you think the flakiness dashboard could show the expected outcome for expects, and possibly not construct a link to the log for the desirable outcome?

Jonas

**Comment by whesse** [comment]
Yes.  We may have to add a field to flaky.json, if it doesn't include the expected outcome, only the actual outcomes.

**Comment by sortie** [comment]
Done: https://dart-review.googlesource.com/c/sdk/+/95024

Once that lands you'll be able to have the dashboard show the desired expectation, and for each outcome know whether they'll have any logs (the maches array).

**Comment by alexmarkov** [comment]
Thank you for fixing this!

**Comment by sortie** [comment]
Alright, flaky.json now contains the data needed. It should now be possible to update the dashboard to show the expected result and not link to flaky correct outcomes.

**Comment by sortie** [comment]
Hi Bill, can you update the dashboard with the new flaky.json data, the 'expected' field and the 'matches' map for whether to link to a log for the given outcome?

**Comment by sortie** [comment]
Ping. Bill can you update the flakiness dashboard to only link to logs for
an outcome if it's wrong? You should also link to flaky passes, if the pass
expectation is wrong.

**Comment by sortie** [comment]
Ping.

**Comment by sortie** [comment]
Ping.
